test/suite/wtthread.py

```diff
@@ -30,17 +30,33 @@
 from helper import compare_tables
 
 class checkpoint_thread(threading.Thread):
-    def __init__(self, conn, done):
+    def __init__(self, conn, done, **kwargs):
+        """
+        Keyword Args:
+            checkpoint_count_max (int): Maximum number of checkpoints to initiate. Must be greater
+                than zero. Thread will not exit until signalled via done.
+        """
         self.conn = conn
         self.done = done
+
+        self.continue_checkpointing = lambda: True
+        
+        if "checkpoint_count_max" in kwargs:
+            count_max = int(kwargs["checkpoint_count_max"])
+            if count_max <= 0:
+                raise ValueError("checkpoint_count_max must be a positive integer")
+            self.__checkpoint_count = 0
+            self.continue_checkpointing = lambda: self.__checkpoint_count < count_max
+
         threading.Thread.__init__(self)
 
     def run(self):
         sess = self.conn.open_session()
-        while not self.done.is_set():
+        while not self.done.is_set() and self.continue_checkpointing():
             # Sleep for 10 milliseconds.
             time.sleep(0.001)
             sess.checkpoint()
+        self.done.wait()
```

Comment:
A comment here to explain why we need wait() would be nice!

Reply:
Removed